src/library/solrize.py

```diff
@@ -210,7 +211,7 @@ def process_hash_list(document_datasets):
                         sub_list_data[element_name] = fa['@'+element_name]
                         del fa['@'+element_name]
 
-                fa['id'] = utils.get_hash_for_identifier(json.dumps(fa))
+                fa['id'] = "{}--{}--{}".format(file_id, hashed_iati_identifier, index)
```

Comment:
The index needs to be per-identifier. For the good case (no dupe id's in data) we have an id like "F1-ID1-0" and the idea is that next time we update we use the same Id so the new record overwrites the old one straight away. Currently index is taken from the flattened_activities position so it will be something like "F1-D1-567" and if an activity at the start of the file is deleted before the next run it will change to "F1-D1-566" and then it won't overwrite.